### PR TITLE
Cargo: Use a separate scope for build dependencies

### LINKED ISSUE #1708: Scopes returned by Cargo analyzer not aligned with Cargo.toml naming
ORT Cargo analyzer currently returns 'devDependencies' for development or build dependencies found which is not in line with Rust's language as in the Cargo.toml file the are defined as 'dev-dependencies' and 'build-dependencies' (see links below). Propose to https://github.com/heremaps/oss-review-toolkit/blob/master/analyzer/src/main/kotlin/managers/Cargo.kt#L71 to 'development-dependencies' and introduce also logic for scope 'build-dependencies'

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies
https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#build-dependencies

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Resolves #1708.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Cargo.kt [hunk 1]
```diff
@@ -69,6 +69,7 @@ class Cargo(
         private const val REQUIRED_CARGO_VERSION = "1.0.0"
         private const val SCOPE_NAME_DEPENDENCIES = "dependencies"
         private const val SCOPE_NAME_DEV_DEPENDENCIES = "devDependencies"
```
Comment (tsteenbe) [inline analyzer/src/main/kotlin/managers/Cargo.kt]: See #1708 we should follow scope naming in Rust lang docs [1][2] and rename "devDependencies" to "dev-dependencies" and  buildDependencies to "build-dependencies".  The scope "devDependencies" name was only introduced as @boxdot based the Rust implementation of the NPM one.


[1] https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies
[2] https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#build-dependencies

File: analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml [hunk 2]
```diff
@@ -19,6 +19,10 @@ project:
     path: "<REPLACE_PATH>"
   homepage_url: "https://example.org"
   scopes:
+  - name: "buildDependencies"
```
Comment (tsteenbe) [inline analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml]: Should be build-dependencies

File: analyzer/src/main/kotlin/managers/Cargo.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Cargo.kt]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/cargo-expected-output.yml]: Done.

**Pull request merged**